sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -49,17 +49,24 @@ bool handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
     }
   }
 
-  size_t VerSize = 0;
-  Plugin.call<PiApiKind::piDeviceGetInfo>(Device, PI_DEVICE_INFO_VERSION, 0,
-                                          nullptr, &VerSize);
+  // Some of the error handling below is special for particular OpenCL
+  // versions.  If this is an OpenCL backend, get the version.
+  bool IsOpenCL = false;    // Backend is any OpenCL version
+  bool IsOpenCLV1x = false; // Backend is OpenCL 1.x
+  bool IsOpenCLV20 = false; // Backend is OpenCL 2.0
   if (Platform.get_backend() == cl::sycl::backend::opencl) {
-    assert(VerSize >= 10 &&
-           "Unexpected device version string"); // strlen("OpenCL X.Y")
+    size_t VersionStringSize = 0;
+    Plugin.call<PiApiKind::piDeviceGetInfo>(Device, PI_DEVICE_INFO_VERSION, 0,
+                                            nullptr, &VersionStringSize);
+    std::unique_ptr<char[]> Version(new char[VersionStringSize]);
```

Comment:
But my point was to avoid dynamic allocation here.
Just `char Version[VersionStringSize];

Reply:
Variable sized arrays like that aren't legal C++, right?  I thought they were only an extension that some compilers provide.

Reply:
Sure, and even in C it is optional, starting with C11.
But you can guess statically the max useful size of it as it will be "OpenCL..." and just read that size max with `Plugin.call<PiApiKind::piDeviceGetInfo>`.
Probably 11 characters is a good guess for now.